jobfunnel/backend/scrapers/base.py

```diff
@@ -423,6 +416,132 @@ def _validate_get_set(self) -> None:
                 [field.name for field in excluded_fields]
             )
 
+    def get_job_soups(self) -> List[BeautifulSoup]:
+        """Scrapes raw data from a job source into a list of job-soups
+
+        Returns:
+            List[BeautifulSoup]: list of jobs soups we can use to make Job init
+        """
+        n_pages = self._get_n_pages()
+
+        # Init list of job soups
+        job_soup_dict = {}  # type: List[Any]
+
+        # Init threads & futures list FIXME: we should probably delay here too
+        threads = ThreadPoolExecutor(max_workers=MAX_CPU_WORKERS)
+        try:
+            # Scrape soups for all the result pages containing many job listings
+            futures = []
+            for page in range(1, n_pages+1):
+                futures.append(
+                    threads.submit(
+                        self._get_job_soups_page, page, job_soup_dict
+                    )
+                )
+
+            # Wait for all scrape jobs to finish
+            wait(futures)
+        finally:
+            threads.shutdown()
+
+        return list(job_soup_dict.values())
+
+    def _get_n_pages(self, max_pages: Optional[int] = None) -> int:
+        """Calculates the number of pages of job listings to be scraped.
+
+        i.e. your search yields 230 results at 50 res/page -> 5 pages of jobs
+
+        Args:
+			max_pages: the maximum number of pages to be scraped.
+        Returns:
+            The number of pages to be scraped.
+        """
+        # Get the html data, initialize bs4 with lxml
+        first_soup = self._get_search_page_soup()
+        num_res, n_pages = self._extract_pages_and_total_listings(first_soup)
+
+        self.logger.debug(f"Found {num_res} job postings resulting in {n_pages} pages")
+
+        if not max_pages:
+            return n_pages
+        elif n_pages < max_pages:
+            return n_pages
+        else:
+            return max_pages
+
+    @abstractmethod
+    def _extract_pages_and_total_listings(self, soup: BeautifulSoup) -> Tuple[int, int]:
```

Comment:
all abstract methods should have a name without the `_`, I had added that to indicate that those methods were private to the specific scraper class. 

additionally all the stubs should have a detailed docstring explaining the expected implementation